rasa/cli/test.py

```diff
@@ -167,11 +168,17 @@ def run_nlu_test(args: argparse.Namespace) -> None:
             args.model, "model", DEFAULT_MODELS_PATH
         )
 
-        test_nlu(model_path, nlu_data, output, vars(args))
+        await test_nlu(model_path, nlu_data, output, vars(args))
 
 
-def test(args: argparse.Namespace):
+async def test_async(args: argparse.Namespace):
     """Run end-to-end tests."""
     setattr(args, "e2e", True)
-    run_core_test(args)
-    run_nlu_test(args)
+    await run_core_test(args)
+    await run_nlu_test(args)
+
+
+def test(
+    args: argparse.Namespace, loop: Optional[asyncio.AbstractEventLoop] = None,
+):
+    return rasa.utils.common.run_in_loop(test_async(args), loop,)
```

Comment:
as discussed in our call I'd move `run_in_loop` further down to something like a `test_nlu_async` and `test_perform_nlu_cross_validation_async`. 

This gives us the benefit of reusable `async` functions. Functions with the `argparse` arguments are not really reusable as only `argparse` passes the arguments within the `argparse.Namespace` object. This should also help you to get rid of the error which you had [here](https://github.com/RasaHQ/rasa/pull/7539/files#r545251989)

Reply:
I'd imagine something like 

test -> run_nlu_test ->  -> `test_nlu` (and keep that async) . Then we actually don't even need something like a `test_nlu_async`

Reply:
You would probably do the `run_in_loop` [here](https://github.com/RasaHQ/rasa/blob/b00ab8e1b0c048811b6a4ebee5b589161b1cd72c/rasa/cli/test.py#L171) then 🤔